core/src/main/java/fj/P2.java

```diff
@@ -1,8 +1,10 @@
 package fj;
 
 import static fj.Function.*;
-
+import static fj.data.optic.PLens.pLens;
```

Comment:
That is a valid concern. However there is already many cyclic packages references in fj (eg. Monoid<->Stream). Scalaz solve cyclic references by putting all in a single package. For those reasons I considered all of fj to be a single 'package'.
I can move optic instance into their own package/artifact, the downside I can see is that it will be harder to find them. pro is that it will address @mperry concern about having a clear distinction for optics 'factory methods'.